noarkexchange/src/main/java/no/difi/meldingsutveksling/noarkexchange/PayloadUtil.java

```diff
@@ -98,6 +108,60 @@ public static String queryPayload(Object payload, String xpath) throws PayloadEx
         return result;
     }
 
+    public static List<NoarkDocument> parsePayloadForDocuments(Object payload) throws PayloadException {
+        List<NoarkDocument> docs = Lists.newArrayList();
+
+        String doc;
+        if (payload instanceof String) {
+            doc = (String) payload;
+        } else {
+            doc = ((Node) payload).getFirstChild().getTextContent().trim();
+        }
+
+        XMLInputFactory xmlInputFactory = XMLInputFactory.newInstance();
+        XMLEventReader eventReader;
+        try {
+            eventReader = xmlInputFactory.createXMLEventReader(new StringReader(doc));
+            NoarkDocument noarkDocument = null;
+            while (eventReader.hasNext()) {
+                XMLEvent event = eventReader.nextEvent();
+
+                if (event.isStartElement()) {
+                    StartElement start = event.asStartElement();
+                    switch (start.getName().getLocalPart()) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS134)